### ISSUE TITLE
Clicking RAlt activates a sticky windows key

### ISSUE THREAD

**Issue by dmuiX** [rbreaves/kinto#397]
**Describe the bug**
Disclaimer: All Keys I am describing here are the original keys not the mapped one.
I am not sure if its a bug, could be a feature, but I haven't found anything about it so far neither in the github repo nor in the kinto.ahk file. If I tip one time on my RAlt key the Windows key gets somehow sticked and I cannot type normal keys anymore as they are always recognized as combinations of Windows + a typed key. For example if I click Space its opening my windows menu. If this beaviour is deactivated I have to use Windows + Space. What I found out so far: I can deactivate this behaviour with LAlt (original key) + two times RAlt key. Looked for RAlt and Windows Key in kinto.ahk but haven't found anything.

Thanks a lot for this though, its working damn good =). I'd really love to use this keymap for autohotkey as I am using privately a Macbook but at work a Windows PC and my brain does not like to switch :D.

**Expected behavior**
I wish that it RAlt would just act as normal RAlt so I can type for example | without having to deactivate this behaviour. With this thing its really anoying...Maybe there is already a solution but I haven't seen/found it yet.

**Install Type:** Bare Metal
**Distro:** Windows 10
**Keyboard Type:** Windows
**Branch:** master
**Commit:** current release from github

**Additional context**
Using a Windows keyboard and keyboard type was to Windows.

**Comment by madshaven** [comment]
Same issue. Particularly annoying on a Norwegian keyboard layout as rALT (alt_gr) is used for making @, usually. I see, however, that pressing rCTRL seems to work as I expected alt_gr to work. So perhaps alt_gr is meant to be sticky?

**Comment by rbreaves** [comment]
A swap is happening but I’ve not seen this issue on a US layout. Will need to do more testing & suspending the app & unsuspecting resolves stuck modifiers.

**Comment by dmuiX** [comment]
Have tried to switch from german to english keyboard configuration. Here is the same issue. Dunno if you meant that with 
> A swap is happening but I’ve not seen this issue on a US layout
Actually I don't have a physical US keyboard, its a german one.

**Comment by dmuiX** [comment]
> suspending the app & unsuspecting resolves stuck modifiers.
You mean it will be gone for good? So if I tipp again on altgr it will not be sticky? I will try that.

**Comment by dmuiX** [comment]
> 
> 
> Same issue. Particularly annoying on a Norwegian keyboard layout as rALT (alt_gr) is used for making @, usually. I see, however, that pressing rCTRL seems to work as I expected alt_gr to work. So perhaps alt_gr is meant to be sticky?

On my german keyboard I have this altgr key and in normal config without kinto its not sticky if I tip it, it reacts as a normal altgr key. So I can use these characters @ | \ } ] [ {. Its very anoying if I am not able to use it.

**Comment by dmuiX** [comment]
> 
> 
> ;> suspending the app & unsuspecting resolves stuck modifiers.
> 
> You mean it will be gone for good? So if I tipp again on altgr it will not be sticky? I will try that.

De

**Comment by dmuiX** [comment]
> 
> 
> > suspending the app & unsuspecting resolves stuck modifiers.
> 
> You mean it will be gone for good? So if I tipp again on altgr it will not be sticky? I will try that.

yeah its gone temporary, but will get sticky if I tip it again. So this is not a solution as I need the alt_gr key to act normal ;).

**Comment by dmuiX** [comment]
As this is not a normal behaviour I suspect you have configured something in the autohotkey script. So it would be amazing if you tell me where I can set the alt_gr to act as a normal alt_gr key. I have looked in the autohotkey script but haven't found anything by myself.

**Comment by rbreaves** [comment]
Sometimes the order and interaction of all the various elements in kinto.ahk can have unexpected impacts - my suggestion would be to pull up a text editor like Sublime Text or VSCode and highlight chunks of code and disable large portions within the virtm code block by pressing Cmd + / . I have had to do that at times to find and resolve possible interaction issues with wordwise related hotkeys - the order in which remaps appear can be surprisingly relevant, so it would not surprise me if Alt_Gr might work under certain circumstance, aka without the entirety of Kinto running.

That would be important to nail down because if that is the case then we can narrow down what specific part of it is causing Alt_Gr to fail and try and resolve that remapping issue from there, either re-order it or change it.

https://www.autohotkey.com/boards/viewtopic.php?t=45435

**Comment by madshaven** [comment]
Could you add to the general description what you intend to map alt_gr and rCtrl to in windows? (physical --> virtual)?

alt_gr is sticky / weird. When I work from Sublime, alt_gr+2 is like a right-click on the second item on my taskbar.
When I use rCtrl from Sublime it seems to be bound to alt+shift. So instead of getting @ from rCtrl+2, like i get here, it changes the layout to two-column-view (i.e. alt+shift+2). 
Now, pressing rCtrl+7, which is not bound to any layout change, does give me a {, as normally alt_gr+7 should.

**Comment by rbreaves** [comment]
Alt_Gr should normally map to the 2nd key to the right of the space bar (cmd would be the 1st). This is optional under Linux already for language reasons mostly, I don't think I implemented the same option as a tweak under windows. Despite that commenting out the keyswap on the right modifiers is literally commenting out 2-3 lines at most. It is possible that remaps might ought to specifically be created for just the left sided hotkeys vs either key, as I am sure there is some mixing going where it could be both or just 1 or the the other atm.

Typically I always use the left side modifiers myself so I know those work in all cases. Also the column shifting is specific to sublime I am sure, just replicating what macOS ought to be doing, may be confusing because some users often get used to certain Linux or Windows behaviors & prefer a mixture of the two. Until I have created a Tweaks menu for Windows users some customizations will be hard to carry over because it is more preference based than anything else.
